src/lib/runner.rs

```diff
@@ -361,6 +361,28 @@ pub(crate) fn run_task_with_options(
             None => (),
         };
 
+        let step = {
```

Comment:
this block needed to be moved up so that expansion happens before condition validation (since condition validation calls the script runner with the args)

Reply:
i get it, but its not really what you want. you just want to expand.
this entire block also updated the current env.
so in case condition fails, the env is still updated. this is unexpected.

Reply:
Ok. Let me fix that.


For the evaluation of the condition script, is the correct behaviour to not use any env information defined in the current task when evaluating the condition for the current task?